### PR TITLE
ref(ui): Add generic to space function

### PR DESCRIPTION
**Pull request by scttcper**
Allows the space function to statically know what the output will be. I'm not sure you can do this with lambda style syntax?

Before
image

After
image

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]
😍 😍 😍 😍 😍

File: src/sentry/static/sentry/app/styles/space.tsx [hunk 1]
```diff
@@ -12,6 +12,8 @@ const SPACES = {
 
 export type ValidSize = keyof typeof SPACES;
 
-const space = (size: ValidSize): typeof SPACES[ValidSize] => SPACES[size];
+function space<Size extends ValidSize>(size: Size): typeof SPACES[Size] {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/styles/space.tsx:15]: I typically try and stick with the single character template variable names, but I'm fine either way.

Do you need the explicit return type?

**Comment by EvanPurkhiser** [comment]
> Allows the space function to statically know what the output will be. I'm not sure you can do this with lambda style syntax?

You can it looks like this:

```jsx
const space = <Size extends ValidSize>(size: Size): typeof SPACES[Size] => SPACES[size];
```

**Review by dashed — APPROVED** [review]

File: src/sentry/static/sentry/app/styles/space.tsx (hunk 1 above)
Comment (scttcper) [reply src/sentry/static/sentry/app/styles/space.tsx]: Looks like typescript does figure out the return type on its own.

**Pull request merged**